### PR TITLE
Support multiple clusters in a region

### PR DESCRIPTION
**Pull request by mirthy**
Potentially support multiple clusters within a region, also support using more than just the k8s single domain.
(support cluster-name.k8s)

### CODE REVIEW

File: scripts/init-variables [hunk 1]
```diff
@@ -14,9 +14,8 @@ AWS_REGION="$1"
 COREOS_CHANNEL="$2"
 COREOS_VM_TYPE="$3"
 
-COREOS_AMI_ID=`curl -s \
-  $(printf "http://%s.release.core-os.net/amd64-usr/current/coreos_production_ami_%s_%s.txt" \
-    $COREOS_CHANNEL $COREOS_VM_TYPE $AWS_REGION)`
+COREOS_AMI_ID=$(curl --silent https://coreos.com/dist/aws/aws-$COREOS_CHANNEL.json | grep -A 2 "$AWS_REGION" | grep $COREOS_VM_TYPE | cut -d"\"" -f4)
+[ -z "$COREOS_AMI_ID" ] && echo "FAIL: region $1 not found or doesn't include a CoreOS $COREOS_CHANNEL, $COREOS_VM_TYPE backed AMI" && exit 1
```
Comment (wellsie) [inline scripts/init-variables:20]: <channel>.release.core-os.net/amd-usr deprecated? I haven't seen any info about that. could you provide a link?

Comment (mirthy) [inline scripts/init-variables:20]: Good catch, I suspect this is unnecessary now, turns out there was something wrong with the AMI (us-west-1, stable. hvm) that CoreOS providing in both the JSON or release.core-os.net.  I'll revert it.  I adapted the JSON parsing from here: https://fak3r.com/2015/07/09/howto-get-a-new-coreos-ami-id/  I mis-read the deprecation of the guy's code and jumping to conclusions about release.core-os.net

Comment (wellsie) [inline scripts/init-variables:20]: cool :)

yeah we are tracking that issue here:
https://github.com/kz8s/tack/issues/31

most regions (if not all) are affected by the looks of it. beta channel looks like it is ok (tested eu-central-1 so far).

thanks for the PR - definitely useful to have that internal TLD parameterized. ty !!

**Comment by wellsie** [comment]
I haven't merged this yet because I am seeing builds fail 50% of time... Continuing to investigate. Will merge once I've sorted out those issues.

**Comment by mirthy** [comment]
From the builds it looks like Circle CI is: Suppressing install of AWS API credentials to /home/ubuntu/.aws/credentials and /home/ubuntu/.aws/config for fork PR builds

**Comment by wellsie** [comment]
CI hasn't ever really done anything useful for this project. It is the manual testing I do daily that showed the issue. Looks like it has to do with route53 dns not resolving in time for etcd spin up. 

Looks like I've been able to solve that issue by terraforming a module at a time.

I'll be pushing those changes shortly - which will incorporate the work from this pr too.

**Comment by mirthy** [comment]
You may want to look at this PR: https://github.com/kz8s/tack/pull/38 to force dependencies between the modules and chain things.

**Pull request closed without merging**